codegen/generate_test.go

```diff
@@ -0,0 +1,67 @@
+package codegen
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRemoteStateConfigToTerraformCode(t *testing.T) {
+	expectedOrdered := []byte(`terraform {
+  backend "ordered" {
+    a = 1
+    b = 2
+    c = 3
+  }
+}
+`)
+	expectedEmpty := []byte(`terraform {
+  backend "empty" {
+  }
+}
+`)
+
+	testCases := []struct {
+		name     string
+		backend  string
+		config   map[string]interface{}
+		expected []byte
+	}{
+		{
+			"remote-state-config-unsorted-keys",
+			"ordered",
+			map[string]interface{}{
+				"b": 2,
+				"a": 1,
+				"c": 3,
+			},
+			expectedOrdered,
+		},
+		{
+			"remote-state-config-sorted-keys",
+			"ordered",
+			map[string]interface{}{
+				"a": 1,
+				"b": 2,
+				"c": 3,
+			},
+			expectedOrdered,
+		},
```

Comment:
golang maps are unordered when they are stored, so changing the order in the call here has no effect. Drop one of the test cases here, and instead update the test routine to call the function N times (20 like the other one?).

Reply:
Changed.